src/App.tsx

```diff
@@ -1,61 +1,119 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { useState } from 'react';
+import { FormData, Todo } from './types';
+import { TodoList } from './components/TodoList';
+import { getNewTodoId } from './services';
+
+const initialForm: FormData = {
+  title: '',
+  userId: 0,
+};
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(todosFromServer);
+  const [formData, setFormData] = useState(initialForm);
+  const [isTitleError, setIsTitleError] = useState(false);
+  const [isUserError, setIsUserError] = useState(false);
+
+  const { title, userId } = formData;
+
+  const isDisabledAdd = !title.trim() || !userId;
+
+  const reset = () => {
+    setFormData({ ...initialForm });
+  };
+
+  const handleAddTodo = (inputData: FormData) => {
+    const newTodo = {
+      ...inputData,
+      id: getNewTodoId(todos),
+      completed: false,
+    };
+
+    setTodos(currentTodos => [...currentTodos, newTodo]);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setFormData(prevData => ({
+      ...prevData,
+      title: event.target.value,
+    }));
+    setIsTitleError(false);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setFormData(prevData => ({
+      ...prevData,
+      userId: +event.target.value,
+    }));
+    setIsUserError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (isDisabledAdd) {
+      setIsTitleError(!title);
+      setIsUserError(!userId);
+
+      return;
+    }
+
+    handleAddTodo(formData);
+    reset();
+  };
```

Comment:
The `handleSubmit` function correctly prevents form submission if validation fails and resets the form upon successful submission. Ensure that the error states are managed properly to provide user feedback.